crates/miden-tx/src/executor/notes_checker.rs

```diff
@@ -215,13 +223,12 @@ where
                         // This combination succeeded; remove the most recently added note from
                         // the remaining set.
                         remaining_notes.remove(idx);
-                        successful_notes = test_notes;
                         break;
                     },
                     _ => {
                         // This combination failed; remove the last note from the test set and
                         // continue to next note.
-                        test_notes.pop();
+                        successful_notes.pop();
                     },
                 }
             }
```

Comment:
@PhilippGackstatter I couldn't find an impl that worked in the loop logic here that involved `try_execute` returning the input notes. But I did get rid of the test_notes clone.

Reply:
Thanks for trying, then let's leave it as-is.